chutney/server-core/src/main/java/com/chutneytesting/server/core/domain/tools/ui/KeyValue.java

```diff
@@ -42,18 +41,19 @@ default String value() {
 
     static List<KeyValue> fromMap(Map<String, String> map) {
         return map.keySet().stream()
-            .map((key) -> ImmutableKeyValue.builder()
+            .map((key) -> (KeyValue) ImmutableKeyValue.builder()
                 .key(key)
                 .value(map.get(key))
-                .build())
-            .collect(Collectors.toList());
+                .build()
+            )
+            .toList();
     }
 
     static Map<String, String> toMap(List<KeyValue> list) {
         return ofNullable(list)
             .map(l -> l.stream()
                 .filter(kv -> StringUtils.isNoneBlank(kv.key()))
-                .collect(Collectors.toMap(KeyValue::key, KeyValue::value, (k1, k2) -> k1, LinkedHashMap::new)))
-            .orElseGet(LinkedHashMap::new);
+                .collect(Collectors.toUnmodifiableMap(KeyValue::key, KeyValue::value, (k1, k2) -> k1)))
+            .orElseGet(Map::of);
```

Comment:
could you double check if there are no regression ?

Reply:
From git history, seemed to be done for component's datasets which do not exist anymore.